### ISSUE TITLE
Group Notifications for Blogs not sending

### ISSUE THREAD

**Issue by klecuyer** [gctools-outilsgc/gcconnex#1668]
## User Story for the issue in the format of "As a user of type AAA, I need XXX so that I can YYY."
A few users have reported not receiving notifications for group content, **specifically blogs**. 

## Details on issue or enhancement
Here's an example:
Username: Elaine.Devine
Link to her notifications (for admins): https://gcconnex.gc.ca/messages/notifications/Elaine.Devine#wb-sm
The group in question: https://gcconnex.gc.ca/groups/profile/233846/engc-apm-working-groupfr
The [last two blogs](https://gcconnex.gc.ca/blog/group/233846/all), published Dec 13 and 18 are not found in her notifications, although other group content (discussions, files, pages) from this group are.

Blogs have been mentioned by multiple users (just today) as not sending notifications. I'll continue to provide more cases here.

## For the development team
- [ ] Issue user story documented
- [ ] UX input received
- [ ] Design completed
- [ ] Design validated by business team / UX
- [ ] Code review completed by peer
- [ ] Issue closing comment references any duplicate or connected issues or pull requests
- [ ] Issue closed

**Comment by klecuyer** [comment]
Another user reporting not receiving notifications for blogs specifically.

She's subcribed to the digest and only gets email notifications so I can't check her site notifications.

Username: Christine.Jean
Note: she is also subscribed to the same group mentioned above and not receiving notifications for blogs. Link to group: https://gcconnex.gc.ca/groups/profile/233846/%7B%22en%22:%22GC%20APM%20Working%20Group%22,%22fr%22:%22%22%7D

**Comment by klecuyer** [comment]
Group notification issue: let me know if you want a separate ticket.

Two users today have reported getting excessive duplicate notifications for bookmarks:
https://gcconnex.gc.ca/messages/notifications/Robert.Muma1 (8 notifications)
https://gcconnex.gc.ca/messages/notifications/Anja.Bilandzic (16 notifications)

It could be that the bookmark editor isn't checking the minor notification box. Also, both bookmarks have since been deleted. It's possible this was user error trying to add a bookmark

**Comment by klecuyer** [comment]
Main ticket: https://github.com/gctools-outilsgc/gcconnex/issues/1227

**Comment by LemieuxGen** [comment]
Reopening this issue. Notifications are not being sent out for blog post. Blogs are used by executives of various levels (including ADM, DMs) to share information and updates to their employees. Many users have raised the issue that they are not receiving notifications when a new blog is posted, which is a big concern. Executives will stop using GCconnex as a means of communication if their employees are not notified. 
@Troy-Lawson @cdegrass Can we prioritize this issue and ensure this is fixed? 
Note that this ticket should not be closed and grouped with the other general notification issue as Blogs is a priority and shouldn't wait until we figure out all the notification issues.
Thanks!

**Comment by klecuyer** [comment]
A list of GCconnex groups reporting this issue specific to blogs. This issue is on GCcollab and GCconnex @JOutram77 @LemieuxGen @Steph4104 @pandurx :
GC APM Working Group: https://gcconnex.gc.ca/groups/profile/233846/gc-apm-working-group
Engaging Indigenous Peoples https://gcconnex.gc.ca/groups/profile/16100881/enengaging-indigenous-peoples-mobilisation-des-peuples-autochtonesfr
SSC Recruitment: https://gcconnex.gc.ca/groups/profile/25961968/enssc-recruitmentfrrecrutement-au-spc
Executives Corner: https://gcconnex.gc.ca/groups/profile/19651377/enexecutive-corner-carrefour-des-cadres-supu00e9rieursfr
The Policy Community (GCcollab): https://gccollab.ca/groups/profile/85014/the-policy-community-la-communaute-des-politiques
Other users have just reported that generally they do not receive notifications for blogs.

**Comment by Steph4104** [comment]
@klecuyer , @LemieuxGen  Can one of you verify with the creator of the blog, if he/she create a draft before and publish the draft after? Because this is how a user can publish without sending notification.

**Comment by Steph4104** [comment]
I also notice, after 1min of none activity, the blog generate an auto-save. This create a draft and, as I say before, notification are not send if a draft is create.

**Comment by klecuyer** [comment]
@Steph4104 that's good to know. I never thought about the auto-save and that's likely the case whenever this happens, or at least one of them. 

Is it possible (or reasonably doable) to make it so when you Publish from a Draft it sends a notification? Or if it saves as a Draft then that turns off notifications for that post for good.

**Comment by Steph4104** [comment]
@klecuyer I can easily extend to 12h before execute the auto-save.

And, I can also take a look at send a notification when a draft is publish, but I'm not sure if user use this way to publish without sending notification.

**Comment by klecuyer** [comment]
@Steph4104 I don't think we want to send a notification when it's saved as a Draft. Only if it's first saved as a Draft (or auto-saved) then Published afterwards. Only sending the notification when the Draft gets Published. Is that possible?
@LemieuxGen

**Comment by Steph4104** [comment]
@klecuyer ,Do we want the draft function? Cause if the user create a draft, and publish after, I can make it send a notification but if the user put it back as draft and publish again, it will send an other notification.

Or, I can remove the draft function.

**Comment by LemieuxGen** [comment]
@Steph4104 @klecuyer We should keep the draft function, but a notification should be sent when the blog is published (going from draft to publish should send a notification).
